crates/graphql/src/schema/objects/marketplace.rs

```diff
@@ -103,4 +105,80 @@ impl Marketplace {
             .await
             .map_err(Into::into)
     }
+
+    #[graphql(arguments(
+        wallet(description = "wallet of current user"),
+        creators(description = "creators of nfts"),
+        listed(description = "addresses of auction house")
+    ))]
+    fn nft_counts(
+        &self,
+        wallet: PublicKey<Wallet>,
+        creators: Vec<PublicKey<Wallet>>,
+        listed: Vec<PublicKey<Wallet>>,
+    ) -> NftCounts {
+        NftCounts::new(wallet, creators, listed)
+    }
+}
+
+#[derive(Debug, Clone)]
+struct NftCounts {
+    wallet: PublicKey<Wallet>,
+    creators: Vec<PublicKey<Wallet>>,
+    listed: Vec<PublicKey<Wallet>>,
+}
+
+impl NftCounts {
+    #[must_use]
+    pub fn new(
+        wallet: PublicKey<Wallet>,
+        creators: Vec<PublicKey<Wallet>>,
+        listed: Vec<PublicKey<Wallet>>,
+    ) -> Self {
+        Self {
+            wallet,
+            creators,
+            listed,
+        }
+    }
+}
+
+#[graphql_object(Context = AppContext)]
+impl NftCounts {
+    fn total(&self, context: &AppContext) -> FieldResult<i32> {
+        let conn = context.db_pool.get()?;
+        let creators = &self
+            .creators
+            .map(|a| a.into_iter().map(Into::into).collect());
+        let count = nft_count::total(&conn, creators)?;
+        Ok(count.try_into()?)
+    }
+
+    fn listed(&self, context: &AppContext) -> FieldResult<i32> {
+        let conn = context.db_pool.get()?;
+        let creators = &self
+            .creators
+            .map(|a| a.into_iter().map(Into::into).collect());
+        let list = &self.listed.map(|a| a.into_iter().map(Into::into).collect());
+        let count = nft_count::listed(&conn, creators, list)?;
+        Ok(count.try_into()?)
+    }
+
+    fn owned(&self, context: &AppContext) -> FieldResult<i32> {
+        let conn = context.db_pool.get()?;
+        let creators = &self
+            .creators
+            .map(|a| a.into_iter().map(Into::into).collect());
+        let count = nft_count::owned(&conn, creators, &self.wallet)?;
+        Ok(count.try_into()?)
```

Comment:
Shouldn't need this try into since already converting the count in the queries from i64 to i32. Actually recommend keeping this try_into but return i64 from the queries since thats the db type. Do conversion in api layer where its enforcing the i32.